index.js

```diff
@@ -454,14 +454,15 @@ function fastifyMultipart (fastify, options = {}, done) {
       const target = createWriteStream(filepath)
       try {
         await pump(file.file, target)
-        this.tmpUploads.push(filepath)
         requestFiles.push({ ...file, filepath })
         // busboy set truncated to true when the configured file size limit was reached
         if (file.file.truncated) {
           const err = new RequestFileTooLargeError()
           err.part = file
           throw err
         }
+        // Delay adding the filepath to tmpUploads until the file has been checked for size 
+        this.tmpUploads.push(filepath)
```

Comment:
We need to add it there to clean up the files after the response.

Reply:
I also opened an issue to explain further why this change is necessary, see #172

Reply:
In short: this adds the file to the list to be cleaned up when the response is closed, but by then it is already deleted.

Reply:
You're right. We remove the file afterward in the catch handler. I propose to leave it as it is but remove the additional `unlink` operation in the catch case.

Reply:
Ok, I'm thinking this should be better